controllers/watcher_controller.go

```diff
@@ -245,6 +253,24 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 		return ctrl.Result{}, errors.New("error retrieving required data from transporturl secret")
 	}
 
+	// Prometheus config secret
+
+	hashPrometheus, _, _, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.PrometheusSecret},
+		[]string{
+			PrometheusHost,
+			PrometheusPort,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		r.RequeueTimeout,
+	)
+	if err != nil || hashPrometheus == "" {
+		// Empty hash means that there is some problem retrieving the key from the secret
+		return ctrl.Result{}, errors.New("error retrieving required data from prometheus secret")
```

Comment:
it might do this already but this should result in the inputready condition being false i think
we could also do something like

```
		instance.Status.Conditions.Set(condition.FalseCondition(
			watcherv1beta1.WatcherPrometheusReadyCondition,
			condition.RequestedReason,
			condition.SeverityWarning,
watcherv1beta1.WatcherPrometheusInvalidSecretMessage))
```

Reply:
by the way i noteest that for the transport URL we are doing

ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil

that a latent bug that we should fix separate.

we should not be doing any requeueing manually.

that one fo the anti-patterns that we need to review for and address in our existing code.

Reply:
> it might do this already but this should result in the inputready condition being false i think

I'll use the regular inputready condition with a custom message for prometheus secret error, is that fine?